include/tunnler/room_member.h

```diff
@@ -111,14 +120,18 @@ class RoomMember final {
     int GetPing() const;
 
     /**
-     * Registers a function to call, when WifiPackets were received
+     * Registers a callback functions to the room member
+     * @param callback The function to call
+     * @param event_type The event on which callback will be called
      */
-    void RegisterOnFrameReceived(std::function<void()> callback) { OnFrameReceived = callback; };
+    void Register(Callback callback, EventType event_type);
```

Comment:
> It might be only me, but I feel this new way of handling callbacks is rather strange and unintuitive.
I preferred your previous approach (which can be handled the new way internally).

Yes. With this PR I've introduced two things: 

- the  vector for adding multiple callbacks for one event
- the enum to add more event types really easy.

With the "old" way we have to create a lot of Register/Derigister functions. I think internally would could handle it the same way or treat every event separate, with an own map, mutex, etc...

> I assume user arguments are going to be passed through std::bind ?

Yes, or no practically but a lambda works, too